src/mpyl/utilities/repo/__init__.py

```diff
@@ -244,52 +249,58 @@ def changes_between(self, base_revision: str, head_revision: str) -> list[Revisi
         changed_files = self._repo.git.diff(
             f"{base_revision}..{head_revision}", name_only=True
         )
-        return Revision.from_git_output(revs, changed_files)
+        return Changeset.from_git_output(revs, changed_files)
 
-    def changes_in_branch(self) -> list[Revision]:
-        base_ref = self.base_revision
-        base_hex = base_ref.hexsha if base_ref else self.root_commit_hex
+    def changes_in_branch(self) -> Changeset:
+        return Changeset(self.get_sha, self.changed_files_in_branch())
 
-        head_hex = self._repo.active_branch.commit.hexsha
-
-        return self.changes_between(base_hex, head_hex)
+    def changed_files_in_branch(self) -> set[str]:
+        # TODO pass the base_branch as a build parameter, not all branches  # pylint: disable=fixme
+        #  are created from the main branch
+        #  Also throw a more specific exception if the base branch is not found
+        base_branch = self.main_origin_branch
+        changed_files = self._repo.git.diff(
+            f"{base_branch}...HEAD", name_only=True
+        ).splitlines()
+        return set(changed_files)
 
-    def changes_in_commit(self) -> set[str]:
+    def files_changed_locally(self) -> set[str]:
         changed: set[str] = set(
             self._repo.git.diff(
                 self.__get_filter_patterns(), None, name_only=True
             ).splitlines()
         )
         return changed.union(self._repo.untracked_files)
 
-    def changes_in_branch_including_local(self) -> list[Revision]:
-        in_branch = self.changes_in_branch()
-        in_branch.append(
-            Revision(len(in_branch), self.get_sha, self.changes_in_commit())
+    def changes_in_branch_including_local(self) -> Changeset:
+        return Changeset(
+            sha=self.get_sha,
+            files_touched=(
+                self.changed_files_in_branch() | self.files_changed_locally()
```

Comment:
I had to look up what `|` is doing in python, I'd just use
```suggestion
                self.changed_files_in_branch().union(self.files_changed_locally())
```

Reply:
We do this all over the place actually 😅 maybe you can make a new pr changing it everywhere if you prefer `.union`?